pages/api/address/[address]/delegated-to.ts

```diff
@@ -6,28 +6,50 @@ SPDX-License-Identifier: AGPL-3.0-or-later
 
 */
 
-import invariant from 'tiny-invariant';
 import { NextApiRequest, NextApiResponse } from 'next';
-import { isSupportedNetwork } from 'modules/web3/helpers/networks';
 import { fetchDelegatedTo } from 'modules/delegates/api/fetchDelegatedTo';
 import { DelegationHistoryWithExpirationDate } from 'modules/delegates/types';
 import BigNumber from 'lib/bigNumberJs';
 import withApiHandler from 'modules/app/api/withApiHandler';
-import { DEFAULT_NETWORK } from 'modules/web3/constants/networks';
+import { DEFAULT_NETWORK, SupportedNetworks } from 'modules/web3/constants/networks';
 import { resolveENS } from 'modules/web3/helpers/ens';
 import { getContracts } from 'modules/web3/helpers/getContracts';
 import { networkNameToChainId } from 'modules/web3/helpers/chain';
 import { getVoteProxyAddresses } from 'modules/app/helpers/getVoteProxyAddresses';
+import { ApiError } from 'modules/app/api/ApiError';
+import { isValidAddressParam } from 'pages/api/polling/isValidAddressParam';
+import validateQueryParam from 'modules/app/api/validateQueryParam';
 
 export type MKRDelegatedToAPIResponse = {
   delegatedTo: DelegationHistoryWithExpirationDate[];
   totalDelegated: number;
 };
 export default withApiHandler(
   async (req: NextApiRequest, res: NextApiResponse<MKRDelegatedToAPIResponse>) => {
-    const network = (req.query.network as string) || DEFAULT_NETWORK.network;
+    // validate network
+    const network = validateQueryParam(
+      (req.query.network as SupportedNetworks) || DEFAULT_NETWORK.network,
+      'string',
+      {
+        defaultValue: null,
+        validValues: [SupportedNetworks.GOERLI, SupportedNetworks.GOERLIFORK, SupportedNetworks.MAINNET]
```

Comment:
i'm thinkingwe could add a "required" flag on the validateQueryParams function to make sure the value exists, and we would get some lines of code through the app

Reply:
I was thinking about this too. Maybe we have another parameter for the error to throw too and the function can throw the if some validation fails. What do you think?